File: BackEndE-Commerce/src/main/java/com/example/BackEndE_Commerce/entity/Category.java
package com.example.BackEndE_Commerce.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Le nom de la catégorie est obligatoire")
    private String name;

    private String description;

    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL)
    private List<Product> products;
}

File: E-CommerceFrontEnd/src/app/models/category.ts
import { Product } from './product';

export interface Category {
  id?: number;              // Optionnel car généré par la DB lors de l'ajout
  name: string;             // Correspond à 'name' en Java
  description?: string;      // Optionnel (peut être null)
  products?: Product[];      // Relation OneToMany
}
